src/ripple/rpc/impl/ShardArchiveHandler.cpp

```diff
@@ -76,9 +230,12 @@ ShardArchiveHandler::add(std::uint32_t shardIndex, parsedURL&& url)
     auto const it {archives_.find(shardIndex)};
     if (it != archives_.end())
         return url == it->second;
```

Comment:
Well, there is double book keeping since the database is meant to be a backup, but not all of the work is duplicated. The database is only accessed when a download request is added and removed. Outside of those to occurrences, there are several references to `archives_` that do not also incur database access. Since the duplication only occurs twice for each download request, I don't think its worth using only the database, losing the convenience of the `std` container.